src/com/lftechnology/java/training/sanish/application/model/dao/EmployeeDao.java

```diff
@@ -0,0 +1,196 @@
+            Employee employee = null;
+            if (rs.next()) {
+                employee = new Employee();
+                employee.setResultSetAttributes(rs);
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return employee;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+        return null;
+    }
+
+    @Override public List<Employee> getAll() {
+        List<Employee> employeesList = new ArrayList<Employee>();
+        try {
+            String query = "SELECT * FROM employees";
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            ResultSet rs = preparedStatement.executeQuery();
+            while (rs.next()) {
+                Employee employee = new Employee();
+                employee.setResultSetAttributes(rs);
+                employeesList.add(employee);
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return employeesList;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return employeesList;
+    }
+
+    @Override public <V> List<Employee> getAll(String condition, V... parameters) {
+        List<Employee> employeesList = new ArrayList<Employee>();
+        try {
+            String query = "SELECT * FROM employees WHERE " + condition;
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            try {
+                for (int i = 0; i < parameters.length; i++) {
+                    preparedStatement.setString(i + 1, parameters[i].toString());
+                }
+            } catch (SQLException e) {
+                LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            }
+
+            ResultSet rs = preparedStatement.executeQuery();
+            while (rs.next()) {
+                Employee employee = new Employee();
+                employee.setResultSetAttributes(rs);
+                employeesList.add(employee);
+            }
+
+            rs.close();
+            preparedStatement.close();
+            return employeesList;
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, Constants.EXCEPTION_ERROR_MSG_LABEL + "{0}", new Object[] { e });
+            DbConnect.dbClose();
+        }
+
+        return employeesList;
+    }
+
+    @Override public <V> int update(String condition, String setString, V... parameters) {
+        try {
+            String query = "UPDATE employees SET " + setString + " WHERE " + condition;
+            PreparedStatement preparedStatement = DbConnect.getDbConnection().prepareStatement(query);
+            try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1141)